File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectGoals.java
package com.slavacom.organizationservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "project_goals")
public class ProjectGoals {
	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	@Column(name = "id", nullable = false)
	private UUID id;

	@Column(name = "project_id", nullable = false)
	private UUID projectId;

	@Column(name = "name", nullable = false)
	private String name;

	@Column(name = "description")
	private String description;

	@Column(name = "target_date")
	private Instant targetDate; // планируемая дата достижения

	@Column(name = "progress", nullable = false)
	private Integer progress = 0; // прогресс в процентах (0-100)

	@Column(name = "is_completed", nullable = false)
	private boolean isCompleted = false;

	@Column(name = "responsible")
	private UUID responsible; // ответственный за цель

	@CreationTimestamp
	@Column(name = "created_at", nullable = false, updatable = false)
	private Instant createdAt;

	@UpdateTimestamp
	@Column(name = "updated_at")
	private Instant updatedAt;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectGoals.kt
package com.slavacom.organizationservice.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "project_goals")
open class ProjectGoals {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.UUID)
    @field:Column(name = "id", nullable = false)
    open var id: UUID? = null

    @field:Column(name = "project_id", nullable = false)
    open var projectId: UUID? = null

    @field:Column(name = "name", nullable = false)
    open var name: String? = null

    @field:Column(name = "description")
    open var description: String? = null

    @field:Column(name = "target_date")
    open var targetDate: Instant? = null

    @field:Column(name = "progress", nullable = false)
    open var progress: Int = 0

    @field:Column(name = "is_completed", nullable = false)
    open var isCompleted: Boolean = false

    @field:Column(name = "responsible")
    open var responsible: UUID? = null

    @field:CreationTimestamp
    @field:Column(name = "created_at", nullable = false, updatable = false)
    open var createdAt: Instant? = null

    @field:UpdateTimestamp
    @field:Column(name = "updated_at")
    open var updatedAt: Instant? = null
}

